src/review-todo.js

```diff
@@ -0,0 +1,212 @@
+let isDate = require("date-fns/isDate");
+
+function todo(todoName = "New Todo") {
+	let title = todoName,
+		description = "",
+		urgency = 0,
+		isDone = false,
+		dueDate = todoDate(),
+		scheduleDate = todoDate(),
+		tags = tagList();
+
+	let todoObj = Object.create(todo.proto);
+	Object.assign(todoObj, {
+		title,
+		description,
+		isDone,
+		urgency,
+		dueDate,
+		scheduleDate,
+		tags,
+	});
+
+	return todoObj;
+}
+
+todo.proto = {};
+
+function project(projectName = "New Project") {
+	let proto = Object.assign(project.proto, { isChild });
+	let projectObj = Object.create(proto);
+
+	let title = projectName,
+		description = "",
+		favorite = false,
+		urgency = 0,
+		isDone = false,
+		dueDate = todoDate(),
+		scheduleDate = todoDate(),
+		tags = tagList(),
+		_todoList = createTodoList(projectObj),
+		todoList = Object.create(_todoList),
+		createTag = tagGenerator("project").bind(projectObj);
+
+	todoList.add = (proj) => {
+		if (
+			Object.getPrototypeOf(proj) === project.proto &&
+			projectObj.isChild(proj)
+		) {
+			throw "Looping projects";
+		}
+		_todoList.add(proj);
+	};
+
+	function isChild(project) {
+		let projects = this.tags
+			.list()
+			.filter((tag) => tag.tagType === "project");
+		if (!projects) return false;
+		if (projects.some((proj) => proj.id === project)) return true;
+		if (projects.some((proj) => proj.id.isChild(project))) return true;
+		return false;
+	}
+
+	Object.assign(projectObj, {
+		title,
+		description,
+		favorite,
+		isDone,
+		urgency,
+		dueDate,
+		scheduleDate,
+		tags,
+		todoList,
+		createTag,
+	});
+
+	return projectObj;
+}
+
+project.proto = {};
```

Comment:
You could also add `isChild` to the project prototype here. 
```
project.proto = {
  isChild: function(otherProject) {
    ... // your existing code should mostly work
  }
}
```

Regardless, if [this article](https://levelup.gitconnected.com/the-javascript-object-paradigm-and-prototypes-explained-simply-e9cb9eaa49aa) is right (I should ideally find that in the docs, but I don't have that in me right now). 
> Setting methods this way (in contrast to using this inside of functions) is especially useful because the method implementation will only occur once instead of every time new is called. This will save memory and increase performance.